packages/arb-util/broadcaster/clientmanager.go

```diff
@@ -271,25 +263,22 @@ func (cm *ClientManager) startWriter(ctx context.Context) {
 			case <-ctx.Done():
 				return
 			case data := <-cm.out:
-				cm.mu.RLock()
-				// Copy list so data can be written to each client without lock held
-				clientList := make([]*ClientConnection, len(cm.clientPtrMap))
-				var i uint64
+				clientDeleteList := make([]*ClientConnection, 0, len(cm.clientPtrMap))
+				cm.mu.Lock()
+				// Lock mutex while writing to channels to ensure items delivered in order
 				for client := range cm.clientPtrMap {
-					clientList[i] = client
-					i++
+					if len(client.out) > MaxSendQueue {
```

Comment:
I don't think this can ever be true since the buffer is size `MaxSendQueue`

Reply:
good catch